libraries/chain/database/block_tasks/comments_cashout_impl.cpp

```diff
@@ -81,7 +81,8 @@ process_comments_cashout_impl::comment_payout_result process_comments_cashout_im
 
         auto parent_author_reward = comment.depth == 0
             ? asset(0, reward_symbol)
-            : (author_reward + children_comments_reward) * SCORUM_PARENT_COMMENT_REWARD_PERCENT / SCORUM_100_PERCENT;
+            : protocol::multiply_asset_by_fractional((author_reward + children_comments_reward),
```

Comment:
I dislike `protocol::multiply_asset_by_fractional` as a free function cuz it is not discoverable. What is we create a type called `fraction` and add `operator*` overloading into `asset` type. The this code will be look like `(author_reward + children_comments_reward) * fraction{SCORUM_PARENT_COMMENT_REWARD_PERCENT, SCORUM_100_PERCENT}`